code/jobs/DeployJob.php

```diff
@@ -96,6 +116,17 @@ public function perform() {
 				throw new \RuntimeException($message);
 			}
 
+			if($backupDataTransfer) {
+				$log->write('Backing up existing data');
+				try {
+					$environment->Backend()->dataTransfer($backupDataTransfer, $log);
+					$this->updateBackupStatus('Finished');
```

Comment:
I'd probably set the status directly on $backupDataTransfer->Status directly here.